packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2807,12 +2812,23 @@ export class ContainerRuntime
 				runtimeBatch,
 			);
 		} else {
+			if (!runtimeBatch) {
+				// The DeltaManager used to do this, but doesn't anymore as of Loader v2.4
+				// Anyone listening to our "op" event would expect the contents to be parsed per this same logic
+				if (
+					typeof messageCopy.contents === "string" &&
+					messageCopy.contents !== "" &&
+					messageCopy.type !== MessageType.ClientLeave
```

Comment:
I'm not sure I like special casing of MessageType.ClientLeave. As a general rule, system ops (including summarizeAck) do not contain contents, while runtime op do / can. I think it would be better not to special case one type of system op here.

Reply:
It's based in preserving this code from like 5 years ago. Whatever reason Tanvir had to add it may still matter - like if there's a trailing Leave op that has some no n-JSON string.

I want to remove contents from the "op" event if not remove it altogether and then it won't matter.